src/local_scheduler/local_scheduler.cc

```diff
@@ -59,6 +59,53 @@ void print_resource_info(const LocalSchedulerState *state,
 #endif
 }
 
+void acquire_resources(LocalSchedulerState *state,
+                       LocalSchedulerClient *worker,
+                       double num_cpus,
+                       double num_gpus) {
+  /* Acquire the CPU resources. */
+  state->dynamic_resources[ResourceIndex_CPU] -= num_cpus;
+  CHECK(worker->cpus_in_use == 0);
+  worker->cpus_in_use += num_cpus;
+
+  /* Acquire the GPU resources. */
+  if (num_gpus != 0) {
+    /* Make sure that the worker isn't using any GPUs already. */
+    CHECK(worker->gpus_in_use.size() == 0);
```

Comment:
Good point, also we may want to allow *drivers* to acquire GPUs as well (see the discussion around this comment https://github.com/ray-project/ray/issues/410#issuecomment-290910211).